bundles/org.openhab.binding.webexteams/src/main/java/org/openhab/binding/webexteams/internal/api/WebexTeamsApi.java

```diff
@@ -0,0 +1,281 @@
+            this.authService.importAccessTokenResponse(response);
+            response = this.authService.refreshToken();
+            logger.debug("Initialized from refreshToken");
+            logger.debug("Token {} of type {} created on {} expiring after {} seconds", response.getAccessToken(),
+                    response.getTokenType(), response.getCreatedOn(), response.getExpiresIn());
+        } catch (OAuthException | OAuthResponseException | IOException e) {
+            logger.error("Failed to import refreshToken", e);
+            return "";
+        }
+        return response.getAccessToken();
+    }
+
+    public String getInitialToken() {
+        try {
+            AccessTokenResponse response = authService.getAccessTokenResponseByAuthorizationCode(this.authCode,
+                    OAUTH_REDIRECT_URL);
+            logger.debug("Token {} of type {} created on {} expiring after {} seconds", response.getAccessToken(),
+                    response.getTokenType(), response.getCreatedOn(), response.getExpiresIn());
+
+            // Need to call this manually here. The authService callback only fires on actual refreshes.
+            onAccessTokenResponse(response);
+            return response.getAccessToken();
+        } catch (OAuthException | IOException | OAuthResponseException e) {
+            logger.error("Failed to get initial token: {}", e.getMessage());
+            throw new NotAuthenticatedException("@text/confErrorInitial", e);
+        }
+    }
+
+    public void createBotOAuthClientService() {
+        String thingUID = handler.getThing().getUID().getAsString();
+        AccessTokenResponse response = new AccessTokenResponse();
+        response.setAccessToken(this.authToken);
+        response.setScope(OAUTH_SCOPE);
+        response.setTokenType("Bearer");
+        response.setExpiresIn(Long.MAX_VALUE); // Bot access tokens don't expire
+        OAuthClientService service = oAuthFactory.createOAuthClientService(thingUID, OAUTH_TOKEN_URL,
+                OAUTH_AUTHORIZATION_URL, "not used", null, OAUTH_SCOPE, false);
+        try {
+            service.importAccessTokenResponse(response);
+        } catch (OAuthException e) {
+            throw new WebexTeamsApiException("Failed to create oauth client with bot token", e);
+        }
+        this.authService = service;
+    }
+
+    @Override
+    public void onAccessTokenResponse(AccessTokenResponse tokenResponse) {
+        // Update access token and refreshToken in config
+        logger.debug("Updating refreshToken");
+        String refreshToken = tokenResponse.getRefreshToken();
+        Configuration configuration = handler.editConfiguration();
+        configuration.put("refreshToken", refreshToken);
+        handler.updateConfiguration(configuration);
+    }
+
+    public void dispose() {
+        String thingUID = handler.getThing().getUID().getAsString();
+        this.oAuthFactory.ungetOAuthService(thingUID);
+    }
+
+    public Person getPerson() {
```

Comment:
Can you please add "throws XXX" with the exceptions can be thrown by your method ?

Reply:
Done.

Reply:
Done.